api/internal/filter/authentication.go

```diff
@@ -29,53 +29,61 @@ import (
 
 func Authentication() gin.HandlerFunc {
 	return func(c *gin.Context) {
-		if c.Request.URL.Path == "/apisix/admin/user/login" ||
-			c.Request.URL.Path == "/apisix/admin/tool/version" ||
-			!strings.HasPrefix(c.Request.URL.Path, "/apisix") {
-			c.Next()
-			return
-		}
-
-		tokenStr := c.GetHeader("Authorization")
-		// verify token
-		token, err := jwt.ParseWithClaims(tokenStr, &jwt.StandardClaims{}, func(token *jwt.Token) (interface{}, error) {
-			return []byte(conf.AuthConf.Secret), nil
-		})
-
+		cookie, _ := conf.CookieStore.Get(c.Request, "oidc")
 		errResp := gin.H{
 			"code":    010013,
 			"message": "request unauthorized",
 		}
 
-		if err != nil || token == nil || !token.Valid {
-			log.Warnf("token validate failed: %s", err)
-			c.AbortWithStatusJSON(http.StatusUnauthorized, errResp)
-			return
-		}
+		if cookie.IsNew {
+			if c.Request.URL.Path == "/apisix/admin/user/login" ||
+				c.Request.URL.Path == "/apisix/admin/tool/version" ||
+				!strings.HasPrefix(c.Request.URL.Path, "/apisix") {
```

Comment:
we should put this block out of the block `if cookie.IsNew {`

Reply:
ok